src/ray/core_worker/core_worker.h

```diff
@@ -62,60 +61,185 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
+  TaskExecutionCallback task_execution_callback;
+  /// Application-language callback to check for signals that have been received
+  /// since calling into C++. This will be called periodically (at least every
+  /// 1s) during long-running operations. If the function returns anything but StatusOK,
+  /// any long-running operations in the core worker will short circuit and return that
+  /// status.
+  std::function<Status()> check_signals;
+  /// Application-language callback to trigger garbage collection in the language
+  /// runtime. This is required to free distributed references that may otherwise
+  /// be held up in garbage objects.
+  std::function<void()> gc_collect;
+  /// Language worker callback to get the current call stack.
+  std::function<void(std::string *)> get_lang_stack;
+  /// Whether to enable object ref counting.
+  bool ref_counting_enabled;
+  /// The number of workers to be started in the current process.
+  int num_workers;
+};
+
+/// Lifecycle management of one or more `CoreWorker` instances in a process.
+///
+/// To start a driver in the current process:
+///     CoreWorkerOptions options = {
+///         WorkerType::DRIVER,             // worker_type
+///         ...,                            // other arguments
+///         1,                              // num_workers
+///     };
+///     CoreWorkerProcess::Initialize(options);
+///
+/// To shutdown a driver in the current process:
+///     CoreWorkerProcess::Shutdown();
+///
+/// To start one or more workers in the current process:
+///     CoreWorkerOptions options = {
+///         WorkerType::WORKER,             // worker_type
+///         ...,                            // other arguments
+///         num_workers,                    // num_workers
+///     };
+///     CoreWorkerProcess::Initialize(options);
+///     ...                                 // Do other stuff
+///     CoreWorkerProcess::StartExecutingTasks();
+///
+/// To shutdown a worker in the current process, return a system exit status (with status
+/// code `IntentionalSystemExit` or `UnexpectedSystemExit`) in the task execution
+/// callback.
+///
+/// If more than 1 worker is started, only the threads which invoke the
+/// `task_execution_callback` will be automatically associated with the corresponding
+/// worker. If you started your own threads and you want to use core worker APIs in these
+/// threads, remember to call `CoreWorkerProcess::SetCurrentThreadWorkerId(worker_id)`
+/// once in the new thread before calling core worker APIs, to associate the current
+/// thread with a worker. You can ontain the worker ID via
+/// `CoreWorkerProcess::GetCoreWorker()->GetWorkerID()`.
+///
+/// If only 1 worker is started (either because the worker type is driver, or the
+/// `num_workers` in `CoreWorkerOptions` is set to 1), all threads will be automatically
+/// associated to the only worker. Then no need to call `SetCurrentThreadWorkerId` in
+/// your own threads.
```

Comment:
Can you comment that the above cases are respectively the default behavior of Java and Python?

Reply:
Done.